core/src/main/java/org/elasticsearch/common/logging/DeprecationLogger.java

```diff
@@ -120,16 +155,31 @@ public void deprecated(String msg, Object... params) {
      * @param params The parameters used to fill in the message, if any exist.
      */
     @SuppressLoggerChecks(reason = "safely delegates to logger")
-    void deprecated(Set<ThreadContext> threadContexts, String message, Object... params) {
-        Iterator<ThreadContext> iterator = threadContexts.iterator();
+    void deprecated(final Set<ThreadContext> threadContexts, final String message, final Object... params) {
+        final Iterator<ThreadContext> iterator = threadContexts.iterator();
 
         if (iterator.hasNext()) {
             final String formattedMessage = LoggerMessageFormat.format(message, params);
-
+            final String warningValue =
+                    String.format(
+                            Locale.ROOT,
+                            WARNING_FORMAT,
+                            formattedMessage,
+                            DateTimeFormatter.RFC_1123_DATE_TIME.format(ZonedDateTime.now(GMT)));
+            assert WARNING_HEADER_PATTERN.matcher(warningValue).matches();
```

Comment:
+1 to keep it as we rely on it for deduping. stronger yet - I think we should verify we extract the value